austation/code/game/materialization.dm

```diff
@@ -0,0 +1,297 @@
+
+/obj/structure/table/mat_shiritori/Destroy()
+	if(active)
+		end_game()
+		return ..()
+	STOP_PROCESSING(SSobj, src)
+	QDEL_LIST(entities)
+	current_player = null
+	players = null
+	knockouts = null
+	return ..()
+
+/obj/structure/table/mat_shiritori/proc/ready_up()
+	if(active || game_starting)
+		return
+	instances++
+	setup_blacklist()
+	players.len = 0
+	knockouts.len = 0
+	time_left = initial(time_left)
+	say("Game starting in [start_time] seconds. Place your hand on the table to play.")
+	INVOKE_ASYNC(src, .proc/start_game)
+
+//  Called after everything is ready
+/obj/structure/table/mat_shiritori/proc/start_game()
+	game_starting = TRUE
+	for(var/i in 1 to start_time)
+		var/countdown = start_time - (i - 1)
+		switch(countdown)
+			if(5)
+				say("Game starting in 5...")
+			if(1 to 4)
+				say("[countdown]...")
+		sleep(10)
+	for(var/mob/living/L as() in players)
+		if(QDELETED(L) || L.stat == DEAD)
+			remove_player(L)
+			continue
+	if(length(players) > 1)
+		if(!atom_list)
+			atom_list = generate_name_list(typesof(/atom/movable))
+		active = TRUE
+		current_player = pick(players)
+		say("[current_player] will start.")
+		to_chat(current_player, "<span class='notice'>It is now your turn.</span>")
+		playsound(src, 'sound/effects/gong.ogg', 100, 0, 5)
+		START_PROCESSING(SSobj, src)
+	else
+		say("Insufficient players")
+	game_starting = FALSE
+
+/obj/structure/table/mat_shiritori/proc/end_game()
+	if(!active)
+		return
+	players.len = 0
+	knockouts.len = 0
+	turns = 0
+	instances--
+	if(instances <= 0)
+		atom_list = null
+		blacklist = null
+	QDEL_LIST(entities)
+	time_left = initial(time_left)
+	current_player = null
+	for(var/mob/living/carbon/human/H in players)
+		to_chat(H, "<span class='info'>The game has ended.</span>")
+		REMOVE_TRAIT(H, TRAIT_PACIFISM, "shiritori")
+	STOP_PROCESSING(SSobj, src)
+	active = FALSE
+
+/obj/structure/table/mat_shiritori/attack_hand(mob/user)
+	if(!ishuman(user))
+		return ..()
+	if(active)
+		to_chat(user, "<span class='info'>You can't join an active game.</span>")
+	if(alert(user, "Would you like to [game_starting ? "join the" : "start a"] game of Materialization Shiritori?", "The gods", "Yes", "No") == "Yes")
+		var/mob/living/carbon/human/H = user
+		if(H.stat == DEAD || !H.mind?.hasSoul)
+			return
+		if(HAS_TRAIT(H, TRAIT_MUTE))
+			to_chat(H, "<span class='warning'>You can't play if you can't speak!</span>")
+			return
+		if(!game_starting)
+			ready_up()
+		else
+			say("[H] has joined.")
+		add_player(H)
+
+/obj/structure/table/mat_shiritori/proc/player_say(datum/source, list/speech_args)
+	if(source == current_player)
+		find_object(speech_args[SPEECH_MESSAGE])
+
+/obj/structure/table/mat_shiritori/proc/find_object(phrase as text)
+	phrase = lowertext(phrase)
+	if(length(current_letter) && phrase[1] != current_letter)
+		visible_message("<span class='warning'>Invalid word, entity must begin with \an <b>[current_letter]</b></span>")
+		return
+	if(findtext(phrase, ".", -1))
```

Comment:
```suggestion
	if(phrase[length(phrase)] == ".")
```

Reply:
As far as I'm aware this is either the exact same speed or slower. It also looks worse :)